### PR TITLE
macOS/watchOS/tvOS support

### PR DESCRIPTION
**Pull request by ricardopereira**
- Added support for macOS, watchOS and tvOS
 - Xcode 10: updated the project with recommended settings
 - Project version is now defined in the project to be in common to all targets.
 - Update Podfile to support other platforms (macOS, etc.)
 - Updated the Example project

### CODE REVIEW

**Comment by ricardopereira** [comment]
@rvi Any news about this? Do you need help maintaining this lib?

**Comment by rvi** [comment]
Thanks for your help @ricardopereira, It seems this is also adding watchOS support and tvOS as well? Can you add in the description what is done in the PR?
Do you know why the tests fail?
I don't use this library anymore and don't have time to put into it unfortunately.

**Comment by ricardopereira** [comment]
@rvi Updated the description. I'll have a look at the CI.

**Comment by ricardopereira** [comment]
@rvi Ready to be reviewed again.

**Comment by ricardopereira** [comment]
@rvi bump 😊

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,7 @@
 language: objective-c
-before_install:
-  - gem install cocoapods
+osx_image: xcode10.2
 before_script:
-  - brew update
-  - brew upgrade xctool && brew cleanup xctool
-  - xctool --version
```
Comment (rvi) [inline .travis.yml:7]: Why did you remove these steps? What happens if cocoapods is not installed  for instance?

Comment (ricardopereira) [reply .travis.yml:7]: Travis macOS environment already has CocoaPods installed. You can see: https://docs.travis-ci.com/user/reference/osx#gems-in-the-global-gem-set

> Gems in the global gem set #
> 
> bundler
> rake
> **cocoapods**

About the **`xctool`**, it's an extension for Apple's `xcodebuild`.
So, why using something from others if we can use `xcodebuild` quite easily, something oficial created by Apple.

Comment (rvi) [reply .travis.yml:7]: I don't remember the details, but it wasn't the case when I created the project, 5 years ago.
And xctool wasn't used anymore, so no problem with that.
